kt/plugins/godot-intellij-plugin/build.gradle.kts

```diff
@@ -0,0 +1,98 @@
+import io.gitlab.arturbosch.detekt.Detekt
+import org.jetbrains.changelog.closure
+import org.jetbrains.changelog.markdownToHTML
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    // Java support
+    id("java")
+    // Kotlin support
+    id("org.jetbrains.kotlin.jvm")
+    // gradle-intellij-plugin - read more: https://github.com/JetBrains/gradle-intellij-plugin
+    id("org.jetbrains.intellij") version "0.6.5"
+    // gradle-changelog-plugin - read more: https://github.com/JetBrains/gradle-changelog-plugin
+    id("org.jetbrains.changelog") version "0.6.2"
+    // detekt linter - read more: https://detekt.github.io/detekt/gradle.html
+    id("io.gitlab.arturbosch.detekt") version "1.14.2"
+    // ktlint linter - read more: https://github.com/JLLeitschuh/ktlint-gradle
+    id("org.jlleitschuh.gradle.ktlint") version "9.4.1"
+}
+
+group = "com.utopia-rise"
+version = "0.0.1"
+
+dependencies {
+    implementation(project(":godot-library"))
+    detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.14.2")
+}
+
+// Configure gradle-intellij-plugin plugin.
+// Read more: https://github.com/JetBrains/gradle-intellij-plugin
+intellij {
+    pluginName = "godot-jvm-idea-plugin"
+    version = "2020.1"
+    type = "IC"
+    downloadSources = true
+    updateSinceUntilBuild = true
+
+    setPlugins(
+        "org.jetbrains.kotlin",
+        "com.intellij.java"
+    )
+}
+
+// Configure detekt plugin.
+// Read more: https://detekt.github.io/detekt/kotlindsl.html
+detekt {
+    config = files("$projectDir/detekt-config.yml")
+    buildUponDefaultConfig = true
+
+    reports {
+        html.enabled = false
+        xml.enabled = false
+        txt.enabled = false
+    }
+}
+
+tasks {
+    patchPluginXml {
+        version(version)
+        sinceBuild("201")
+        untilBuild("203.*")
```

Comment:
This will eventually bite you, significant code changes can happen between those versions - then you will eventually need to provide to artifacts one before the significant change and the other after. As I mentioned above, I would suggest to have a release per minor version, so for this part: sinceBuild = 201, untilBuild = 201.* (this means any 201 version)

Reply:
Hmm didn't think of that. I just used the official jetbrains template and this is configured by it like this. I'll take a look at your spek framework plugin and change this here.

Reply:
@raniejade i changed it now to what i think matches what you did in the spek framework plugin. Could you check my impl if this holds up?